test/unit/org/apache/cassandra/config/YamlConfigurationLoaderTest.java

```diff
@@ -27,14 +27,89 @@
+                                                                        .put("client_encryption_options.cipher_suites", Collections.singletonList("FakeCipher"))
+                                                                        .put("client_encryption_options.optional", false)
+                                                                        .put("client_encryption_options.enabled", true)
+                                                                        .build();
+        Config updated = YamlConfigurationLoader.updateFromMap(map, true, config);
+        assert updated == config : "Config pointers do not match";
+        assertThat(config.storage_port).isEqualTo(123);
+        assertThat(config.commitlog_sync).isEqualTo(Config.CommitLogSync.batch);
+        assertThat(config.seed_provider.class_name).isEqualTo("org.apache.cassandra.locator.SimpleSeedProvider");
+        assertThat(config.client_encryption_options.cipher_suites).isEqualTo(Collections.singletonList("FakeCipher"));
+        assertThat(config.client_encryption_options.optional).isFalse();
+        assertThat(config.client_encryption_options.enabled).isTrue();
+    }
+
+    @Test
+    public void withSystemProperties()
+    {
+        // for primitive types or data-types which use a String constructor, we can support these as nested
+        // if the type is a collection, then the string format doesn't make sense and will fail with an error such as
+        //   Cannot create property=client_encryption_options.cipher_suites for JavaBean=org.apache.cassandra.config.Config@1f59a598
+        //   No single argument constructor found for interface java.util.List : null
+        // the reason is that its not a scalar but a complex type (collection type), so the map we use needs to have a collection to match.
+        // It is possible that we define a common string representation for these types so they can be written to; this
+        // is an issue that SettingsTable may need to worry about.
+        try (WithProperties ignore = new WithProperties(CONFIG_ALLOW_SYSTEM_PROPERTIES.getKey(), "true",
+                                                        SYSTEM_PROPERTY_PREFIX + "storage_port", "123",
+                                                        SYSTEM_PROPERTY_PREFIX + "commitlog_sync", "batch",
+                                                        SYSTEM_PROPERTY_PREFIX + "seed_provider.class_name", "org.apache.cassandra.locator.SimpleSeedProvider",
+//                                                        PROPERTY_PREFIX + "client_encryption_options.cipher_suites", "[\"FakeCipher\"]",
+                                                        SYSTEM_PROPERTY_PREFIX + "client_encryption_options.optional", "false",
+                                                        SYSTEM_PROPERTY_PREFIX + "client_encryption_options.enabled", "true",
+                                                        SYSTEM_PROPERTY_PREFIX + "doesnotexist", "true"
```

Comment:
2 nits:

1.) What do you think about a helper in `YamlConfigurationLoader`, perhaps called `asSystemConfigProperty(String)` that applies `SYSTEM_PROPERTY_PREFIX` for us?

2.) Might be a little tricky, given what `WithProperties#close()` does, but should we revert the system property changes before the next test?

Reply:
1) given that only this one tests leverages it, find it over kill
2) thats what `WithProperties` does, it rolls back to the previous state